app/models/order.rb

```diff
@@ -11,17 +11,26 @@ def grandtotal
     item_orders.sum('price * quantity')
   end
 
-  def self.sort_status 
+  def merchant_grandtotal(id)
+    items.where(merchant_id: id).sum('items.price * quantity')
+  end
+
+  def self.sort_status
     self.all.order("status = 1 DESC, status = 0 DESC, status = 2 DESC, status = 3 DESC")
   end
-  
+
   def total_item_quantity
     item_orders.sum(:quantity)
   end
 
+  def merchant_order_items_count(id)
+    items.where(merchant_id: id).sum(:quantity)
+  end
```

Comment:
`items.where(merchant_id: id)` filters out other merchants' stuff from the results before making calculations

Reply:
nice ActiveRecord here!